Optionally.Tests/OptionTests/WhenMatching.cs

```diff
@@ -0,0 +1,62 @@
+﻿using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Optionally.Tests.OptionTests
+{
+    [TestFixture]
+    public class WhenMatching
+    {
+        [Test]
+        public void AndSomeThenSomeFunctionIsCalled()
+        {
+            int expected = 4;
+            var some = Option.Some(4);
+
+            var value = some.Match(
+                ifNone: () => 0,
+                ifSome: x => x);
+
+            Assert.AreEqual(expected, value);
+        }
+
```

Comment:
Total nitpick, but we should think about adding another test that made sure that the function was actually called. At this point, all we know is that a value was returned, but we don't know if the value was hard-coded. A way we can achieve that style of testing looks a lot like below
```
bool someWasCalled = false;
var some = Option.Some(4);

var value = some.Match(
     ifNone: () => 0,
     ifSome: x => {someWasCalled=true; return x;});

Assert.That(someWasCalled);
```

Reply:
Are you wanting both, or use this to replace the existing?

Reply:
Tests now check that function ran, instead of return value.